plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DefaultJdbcMetadata.java

```diff
@@ -1323,13 +1330,98 @@ public ConnectorMergeTableHandle beginMerge(ConnectorSession session, ConnectorT
         RemoteTableName remoteTableName = handle.getRequiredNamedRelation().getRemoteTableName();
 
         List<JdbcColumnHandle> columns = jdbcClient.getColumns(session, schemaTableName, remoteTableName);
-        JdbcOutputTableHandle jdbcOutputTableHandle = (JdbcOutputTableHandle) beginInsert(
-                session,
-                new JdbcTableHandle(schemaTableName, remoteTableName, Optional.empty()),
-                ImmutableList.copyOf(columns),
-                retryMode);
 
-        return new JdbcMergeTableHandle(handle, jdbcOutputTableHandle, primaryKeys, columns, updateColumnHandles);
+        setMergeRollback(new MergeRollbackAction());
```

Comment:
why do we need to explicitly set it. 
Just expose `appendMergeRollbackAction` which will create ate the object on the first call.
Then use `appendMergeRollbackAction` instead of `mergeRollbackAction.get().appendRollback(`